src/reactviews/pages/ExecutionPlan/executionPlanPage.tsx

```diff
@@ -33,10 +34,17 @@ const useStyles = makeStyles({
     },
 });
 
-export const ExecutionPlanPage = () => {
+interface ExecutionPlanPageProps {
+    context?: QueryResultState;
+}
+
+export const ExecutionPlanPage: React.FC<ExecutionPlanPageProps> = ({
+    context,
+}) => {
     const classes = useStyles();
-    const provider = useContext(ExecutionPlanContext);
-    const loadState = provider?.state?.loadState ?? ApiStatus.Loading;
+    const provider = context ? context : useContext(ExecutionPlanContext);
```

Comment:
after this, you can probably just do

```
if (provider === undefined)
return undefined;
```

and avoid many of the null propagation `?`s below